src/lib/loaders/index.ts

```diff
@@ -61,25 +61,35 @@ export interface LoaderFactory {
   ): DynamicPathLoader<BodyAndHeaders<Body, Headers>, PathGeneratorParams>
 }
 
+type loadersAtRoot = LoadersWithoutAuthentication &
+  Partial<LoadersWithAuthentication>
+
+interface availableLoaders extends loadersAtRoot {
+  unauthenticatedLoaders: LoadersWithoutAuthentication
+  authenticatedLoaders: Partial<LoadersWithAuthentication>
+}
+
 /**
  * Creates a new set of data loaders for all routes. These should be created for each GraphQL query and passed to the
  * `graphql` query execution function.
  *
  * Only if credentials are provided will the set include authenticated loaders, so before using an authenticated loader
  * it would be wise to check if the loader is not in fact `undefined`.
  */
-export default (
-  accessToken,
-  userID,
-  opts
-): LoadersWithoutAuthentication & Partial<LoadersWithAuthentication> => {
+export default (accessToken, userID, opts): availableLoaders => {
   const loaders = createLoadersWithoutAuthentication(opts)
+  let authenticatedLoaders = {}
   if (accessToken) {
-    return Object.assign(
-      {},
-      loaders,
-      createLoadersWithAuthentication(accessToken, userID, opts)
+    authenticatedLoaders = createLoadersWithAuthentication(
+      accessToken,
+      userID,
+      opts
     )
   }
-  return loaders
+  return {
+    ...loaders,
+    ...authenticatedLoaders,
+    unauthenticatedLoaders: loaders,
```

Comment:
Why spread `...loaders` up above?

Reply:
Because all the loaders need to exist in the root here, ie- `{ ...loaders }`

Reply:
Yup, this is good 👍

One consistency nit-pick is to rename `loaders` to `unauthenticatedLoaders`.

Reply:
good call!